### ISSUE TITLE
Ask user to insert a device name on first launch instead of generating a random one

### ISSUE THREAD

**Issue by mijorus** [localsend/localsend#977]
### Is your feature request related to a problem? Please describe.

No

### Describe the solution you'd like

A pop-up welcome screen asking the user to provide a name for the current device

**Comment by sergd88** [comment]
I disagree that randomly-generated names are quite ugly. Rather, they just appear somewhat informal and less professional. However, @Tienisto mentioned somewhere that in this regard, he's somewhat inspired by the approach of [Snapdrop](https://snapdrop.net/). [ShareDrop](https://www.sharedrop.io/), for example, uses animal icons. In general, in such cases, I recall Ubuntu, which, despite everything, remains the most popular Linux distribution and is used by many business clients. Their approach to release names, such as **Disco Dingo** 😂, reminds that it isn't necessary to always be so formal 😉

As for a pop-up welcome screen asking user to provide a name for the current device, it's certainly a great idea, though not a new one. The key isn't to greet the user immediately with a name input field 😄

**Comment by mijorus** [comment]
> I disagree that randomly-generated names are quite ugly. Rather, they just appear somewhat informal and less professional. However, @Tienisto mentioned somewhere that in this regard, he's somewhat inspired by the approach of [Snapdrop](https://snapdrop.net/). [ShareDrop](https://www.sharedrop.io/), for example, uses animal icons. In general, in such cases, I recall Ubuntu, which, despite everything, remains the most popular Linux distribution and is used by many business clients. Their approach to release names, such as **Disco Dingo** 😂, reminds that it isn't necessary to always be so formal 😉
> 
> As for a pop-up welcome screen asking user to provide a name for the current device, it's certainly a great idea, though not a new one. The key isn't to greet the user immediately with a name input field 😄

With "ugly", I meant unprofessional.

Ubuntu is not a tool that an average user might use, but rather it's designed for geeks who are used to these kinds of jokes.

While I agree that a pop-up window might not be the best UX, I would at least try to generate a good-enough name based on real device information.

On android, we could use the device name (the same that appears under Bluetooth settings for example)

**Comment by sergd88** [comment]
> > I disagree that randomly-generated names are quite ugly. Rather, they just appear somewhat informal and less professional. However, @Tienisto mentioned somewhere that in this regard, he's somewhat inspired by the approach of [Snapdrop](https://snapdrop.net/). [ShareDrop](https://www.sharedrop.io/
> 
> These are browser-based apps and they are limited by javascript APIs, which does not return any kind of information about the device you are using.
> 
> This is a good this of course, otherwise it would be misused by every advertisment company to track you, but (I Think) a native app should have access to more accurate devide information

I don't think I mentioned the limitations in the JavaScript API anywhere. So, this doesn't relate to the essence of my response at all. I simply explained why the author chose such an implementation and what considerations guided him when assigning, as you noted, "quite ugly" names. That's all.

**Comment by sergd88** [comment]
> Actually, I am also fine if the device name is used in the first place.
> 
> I guess the random generated name commonly used in other file sharing apps is to ensure more privacy because your device might be called "Alice's iPhone".
> 
> An option would be to make the random generated name opt-in for privacy oriented users. There will be a button to generate a random name.

I also don't quite agree with such strict privacy measures, given that many of us don't even turn off Wi-Fi and Bluetooth because they are needed for connected wearable devices and many other applications. But as a developer, I believe you have a better perspective on this.

Considering the above-mentioned fact, the current implementation is probably the most optimal one. Certainly, for those who deploy an entire network for sharing based on LocalSend, it creates certain inconveniences. However, an input window for the name during the first launch of the application probably won't resolve the situation either. And, moreover, it would be a poor decision from a UX perspective.

As for a more formal generated names, currently, the application reads the label for the device itself, such as Windows, Linux, iPhone, Xiaomi, OnePlus, etc. In principle, names can be generated using this label, adding a number to them since there might be several devices with the same label in most cases. Then we would have: iPhone 5, Xiaomi 11, Windows 8, and so on. Since the number immediately after the device label may indicate the version of the device or system, we can separate the label and the number using some constant, like Device. So, the output would be: iPhone Device 5, Xiaomi Device 11, Windows Device 8. For more visual appeal, we could also add a # sign. iPhone Device # 5, Xiaomi Device # 11, Windows Device # 8 (certainly, without gaps between # and the number, which I added due to Markdown formatting). As for the numbers, we could take them from the IP address, although with reconnects, the IP address isn't static, so another device may eventually get the same numeric label. However, the same situation applies to a random names list that repeats. So, this aspect is debatable.

Therefore, the name formula would be: Label + *Device + **Number. *Device = a constant; **Number = random from the list or from the IP address. Of course, this won't help much in differentiating between devices when deploying the app on large networks, but it'll allow for more formal initial names, as I can see the current random names aren't to everyone's liking.

**Comment by mijorus** [comment]
> Since you've already provided an example with Bluetooth, when you first activate the device, you won't be prompted to enter a name for Bluetooth either. Instead, a default name is used, which you can simply change in the Bluetooth settings. Similarly, in the case of LocalSend. The default name is selected from a separately created list, not based on the device's name. As the author has already pointed out, this implementation is related to security.

I've updated the title, to better reflect the goal of this discussion.

Regarding your answer:
- On iPhones, well, it kind of does, considering that usually they are called with your name. 
- While Android phones or windows devices don't show up with your name, they definitely don't generate a random funny name

**Comment by sergd88** [comment]
> Regarding your answer:
> 
> * On iPhones, well, it kind of does, considering that usually they are called with your name.
> * While Android phones or windows devices don't show up with your name, they definitely don't generate a random funny name

Yes, most likely the name for Bluetooth is formed based on your name, but you don't set it directly. It's probably derived from the data you provide for the "Name and Birthdate" item. But, less on that, it's not important at all.

"They definitely don't generate a random funny name." Well, I explained where this idea came from, and as the author said, many other apps/services use a similar approach. Some people like it, some don't, and some are indifferent because you can always open the settings and set the desired name right away. I've a fairly neutral position on this because I know that many people like this option. Although personally, I prefer a more formal approach.

> Yes, that's more or less what I had in mind. Free to agree or disagree. Thank you

This is the only feasible option to ensure a formal approach to the initial device name, which comes to my mind, considering that we cannot read the device name directly.

**Comment by Fred-IV** [comment]
PLEASE, when installing, name it the same as the **Computer Name**. Random names make no sense.
